scripts/entity/base_entity.gd

```diff
@@ -16,11 +16,23 @@ func _ready() -> void:
 
 	# if the movement component and input component exist connect the movement together.
 	if movement_component and input_component and attack_component:
-		input_component.on_horizontal_movement_input.connect(movement_component.on_movement_input)
+		input_component.on_horizontal_movement_input.connect(movement_component.on_horizontal_movement_input)
+		input_component.on_vertical_movement_input.connect(movement_component.on_vertical_movement_input)
 		input_component.on_jump_input_started.connect(movement_component.on_jump_input_started)
 		input_component.on_jump_input.connect(movement_component.on_jump_input)
 		input_component.on_jump_input_cancelled.connect(movement_component.on_jump_input_cancelled)
 		movement_component.on_direction_changed.connect(attack_component.update_direction_specific_items)
 		input_component.on_attack_input_started.connect(attack_component.on_attack_input_started)
 		input_component.on_attack_input.connect(attack_component.on_attack_input)
 		input_component.on_attack_input_cancelled.connect(attack_component.on_attack_input_cancelled)
+		
+
+
+
+func _on_entered_climbable(body: Node) -> void:
+	print("Entered Climbable signal")
+	movement_component._on_entered_climbable()
+
+func _on_exited_climbable(body: Node) -> void:
+	print("Exited Climbable signal")
+	movement_component._on_exited_climbable()
```

Comment:
this should be in the movement component

Reply:
fixed